### PR TITLE
Fix bug in apiserver service cidr split

### PR DESCRIPTION
**Pull request by darshanime**
**What type of PR is this?**
/kind bug

**What this PR does / why we need it:**
Fix bug where kube-apiserver would fail to start when not providing service CIDR

**Which issue(s) this PR fixes:**
NA

**Special notes for your reviewer:**
The `strings.Split` documentation [states](https://golang.org/search?q=strings.Split#Global):
```
// If s does not contain sep and sep is not empty, Split returns a
// slice of length 1 whose only element is s.
```

```package main

import (
	"fmt"
	"strings"
)

func main() {
	serviceClusterIPRanges := ""
	serviceClusterIPRangeList := strings.Split(serviceClusterIPRanges, ",")
	fmt.Printf("len: %d, value: %+v", len(serviceClusterIPRangeList), serviceClusterIPRangeList)

}

// Output
// len: 1, value: []
```

[playground link](https://play.golang.org/p/oLHHJueKmG3)

This would fail the [if check here](https://github.com/kubernetes/kubernetes/pull/85937/files#diff-36e1d7088209b1ab6ff878b18bb35210R586) and apiserver wouldn't start. This PR fixes this bug.

**Does this PR introduce a user-facing change?:**
NA

**Release note:**
```release-note
NONE
```

### CODE REVIEW

File: cmd/kube-apiserver/app/server.go [hunk 1]
```diff
@@ -572,8 +572,12 @@ func Complete(s *options.ServerRunOptions) (completedServerRunOptions, error) {
 
 	// process s.ServiceClusterIPRange from list to Primary and Secondary
 	// we process secondary only if provided by user
-
-	serviceClusterIPRangeList := strings.Split(s.ServiceClusterIPRanges, ",")
+	var serviceClusterIPRangeList []string
```
Comment (aojea) [inline cmd/kube-apiserver/app/server.go]: nit, I think that we can simplify this block with something like this
```
	serviceClusterIPRangeList := []string{}
	if serviceClusterIPRanges != "" {
		serviceClusterIPRangeList = strings.Split(serviceClusterIPRanges, ",")
	}
```

Comment (darshanime) [reply cmd/kube-apiserver/app/server.go]: yes, made the change.

**Comment by aojea** [comment]
thanks
/lgtm

**Comment by darshanime** [comment]
test failures seem to not be related to this change
/retest

**Comment by justaugustus** [comment]
This appears to be a critical fix, which should probably also get cherry-picked to the `release-1.17` branch.

Thoughts Jordan?

/assign @liggitt 
/milestone v1.17
/remove-priority important-soon
/priority critical-urgent

**Comment by justaugustus** [comment]
@darshanime -- This needs a [release note](https://git.k8s.io/community/contributors/guide/release-notes.md).

**Comment by liggitt** [comment]
Also needs a test

**Comment by darshanime** [comment]
@liggitt, I have removed the default `ServiceClusterIPRanges` from the `testServer` used thruout the codebase since it is deprecated. 
Everything passes still, is this enough? Or should I write a separate test for the `app.Complete` func?

**Comment by liggitt** [comment]
Don't change the integration test setup.

It would be better to have the `ServiceClusterIPRanges -> serviceIP, serviceIPRange, secondaryServiceClusterCIDR, error` logic in a standalone function that could be unit tested easily, and then used like this in Complete():
```go
serviceIP, serviceIPRange, secondaryServiceIPRange, err := getServiceIPAndRanges(s.ServiceClusterIPRanges)
if err != nil {
  return options, err
}
s.PrimaryServiceClusterIPRange = serviceIPRange
s.SecondaryServiceClusterIPRange = secondaryServiceIPRange
```

**Comment by darshanime** [comment]
@liggitt thanks for the review. Please take a look now.

**Comment by liggitt** [comment]
need to run `hack/update-bazel.sh` to pick up the new test file:
```diff
diff --git a/cmd/kube-apiserver/app/BUILD b/cmd/kube-apiserver/app/BUILD
index 66570c567e5..95162cf8760 100644
--- a/cmd/kube-apiserver/app/BUILD
+++ b/cmd/kube-apiserver/app/BUILD
@@ -1,4 +1,4 @@
-load("@io_bazel_rules_go//go:def.bzl", "go_library")
+load("@io_bazel_rules_go//go:def.bzl", "go_library", "go_test")
 
 go_library(
     name = "go_default_library",
@@ -99,3 +99,9 @@ filegroup(
     tags = ["automanaged"],
     visibility = ["//visibility:public"],
 )
+
+go_test(
+    name = "go_default_test",
+    srcs = ["server_test.go"],
+    embed = [":go_default_library"],
+)

```

lgtm otherwise, much better tested, thanks!

File: cmd/kube-apiserver/app/server_test.go [hunk 2]
```diff
@@ -0,0 +1,56 @@
+/*
+Copyright 2018 The Kubernetes Authors.
```
Comment (liggitt) [inline cmd/kube-apiserver/app/server_test.go]: 2019 in new files

**Comment by liggitt** [comment]
comments resolved in https://github.com/kubernetes/kubernetes/pull/85968

**Comment by liggitt** [comment]
/close
in favor of #85968

**Comment by darshanime** [comment]
Hi @liggitt, sorry I was asleep, it was already 1am IST here 😴
Thanks for reviewing this 🙏

**Pull request closed without merging**